HDT Quad sink (#32)
SolidBench/rdf-dataset-fragmenter.js: README.md

* `"fileExtension"`: An optional extension to add to resulting files.
* `"iriToPath"`: A collection of mappings that indicate what URL patterns should be translated into what folder structure.

#### HDT Quad Sink

A quad sink that writes to files using an IRI to local file system path mapping and then converts the files into an [HDT document](https://www.rdfhdt.org/what-is-hdt/).
The implementation uses the [docker](https://www.docker.com/) image [HDT-Docker](https://github.com/rdfhdt/hdt-docker) of the [hdt-cpp](https://github.com/rdfhdt/hdt-cpp) library.
The docker operations to acquire the image and execute the transformations into HDT are performed by the sink.

**WARNING: Can be very slow for many files**

```json
{
  "quadSink": {
    "@type": "QuadSinkHdt",
    "log": true,
    "outputFormat": "application/n-quads",
    "fileExtension": "$.nq",
    "iriToPath": {
      "http://example.org/base/": "output/base/",
      "http://example.org/other/": "output/other/"
    },
    "poolSize": 1,
    "deleteSourceFiles": false,

>     "errorFileDockerRfdhdt": "./error_log_docker_rfdhdt"

Reviewer: ```suggestion
    "errorFileDockerRfdhdt": "./error_log_docker_rfdhdt.txt"
```

* `"fileExtension"`: An optional extension to add to resulting files.
* `"iriToPath"`: A collection of mappings that indicate what URL patterns should be translated into what folder structure.

#### HDT Quad Sink

A quad sink that writes to files using an IRI to local file system path mapping and then converts the files into an [HDT document](https://www.rdfhdt.org/what-is-hdt/). 
The implementation uses the [docker](https://www.docker.com/) image [HDT-Docker](https://github.com/rdfhdt/hdt-docker) of the [hdt-cpp](https://github.com/rdfhdt/hdt-cpp) library.
The docker operations to acquire the image and execute the transformations into HDT are performed by the sink. 

> **Can be very slow if there are multiple files (in the range of 10,000) generated**

Reviewer: ```suggestion
**WARNING: Can be very slow for many files**
```

```json
{
  "quadSink": {
    "@type": "QuadSinkHdt",
    "log": true,
    "outputFormat": "application/n-quads",
    "fileExtension": "$.nq",
    "iriToPath": {
      "http://example.org/base/": "output/base/",
      "http://example.org/other/": "output/other/"
    },

>     "hdtConversionOpPoolSize": 5,

Reviewer: ```suggestion
    "poolSize": 5,
```

    "deleteSourceFiles": true
  }
}
```

Options:
* `"log"`: If a quad counter should be shown to show the current progress.
* `"outputFormat"`: The desired output serialization. (Only `"application/n-quads"` is considered stable at the moment).
* `"fileExtension"`: An optional extension to add to resulting files.
* `"iriToPath"`: A collection of mappings that indicate what URL patterns should be translated into what folder structure.

> * `"hdtConversionOpPoolSize"`: The number of concurrent HDT conversion operations.

Reviewer: Can this be an optional param that defaults to `1`?

Reviewer: Let's then also document it as such.

> * `"deleteSourceFiles"`: If the sink should delete the source RDF file after the conversion into HDT.

Reviewer: We probably want this one to default to `false`.

The implementation uses the [docker](https://www.docker.com/) image [HDT-Docker](https://github.com/rdfhdt/hdt-docker) of the [hdt-cpp](https://github.com/rdfhdt/hdt-cpp) library.
The docker operations to acquire the image and execute the transformations into HDT are performed by the sink. 

**WARNING: Can be very slow for many files**

```json
{
  "quadSink": {
    "@type": "QuadSinkHdt",
    "log": true,
    "outputFormat": "application/n-quads",
    "fileExtension": "$.nq",
    "iriToPath": {
      "http://example.org/base/": "output/base/",
      "http://example.org/other/": "output/other/"
    },
    "poolSize": 1,
    "deleteSourceFiles": false,
    "errorFileDockerRfdhdt": "./error_log_docker_rfdhdt"
  }
}
```

Options:
* `"log"`: If a quad counter should be shown to show the current progress.
* `"outputFormat"`: The desired output serialization. (Only `"application/n-quads"` is considered stable at the moment).
* `"fileExtension"`: An optional extension to add to resulting files.
* `"iriToPath"`: A collection of mappings that indicate what URL patterns should be translated into what folder structure.

> * `"poolSize"`: The number of concurrent HDT conversion operations. By the default `1`.

Reviewer: ```suggestion
* `"poolSize"`: The number of concurrent HDT conversion operations. By default `1`.
```

* `"deleteSourceFiles"`: If the sink should delete the source RDF file after the conversion into HDT.

> * `"errorFileDockerRfdhdt"`: File where the error of HDT-Docker will be outputed. By default `"./error_log_docker_rfdhdt"`.

Reviewer: ```suggestion
* `"errorPath"`: File where the error of HDT-Docker will be outputed. By default `"./error_log_docker_rfdhdt"`.
```

Reviewer: Can we make the default `error_log_docker_rfdhdt.txt`?
